src/Hl7.Fhir.Specification/Validation/Validator.cs

```diff
@@ -490,12 +493,16 @@ private OperationOutcome snapshotGenerationNeeded(StructureDefinition definition
             if (generator != null)
             {
                 generator.Update(definition);
+
 #if DUMP_SNAPSHOTS
 
-                string xml = FhirSerializer.SerializeResourceToXml(definition);
-                string name = definition.Id ?? definition.Name.Replace(" ", "");
+                string xml = (new FhirXmlSerializer()).SerializeToString(definition);
+                string name = definition.Id ?? definition.Name.Replace(" ", "").Replace("/", "");
+                var dir = @"c:\temp\validation";
```

Comment:
Hmmm, would this not be better?
`var dir = Path.Combine(Path.GetTempPath(), "validation");`

Reply:
Yes, great idea, will do.  It's only run for debugging, but still, this is placed within production code, so it should be portable.